File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithCheckpointEvent.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl
import com.ing.baker.recipe.scaladsl.Recipe
import examples.scala.events.{FraudCheckCompleted, PaymentReceived}

object RecipeWithCheckpointEvent {
  val recipe: Recipe = Recipe("example")
    .withCheckpointEvent(
      scaladsl.CheckPointEvent(
        name = "CheckpointReached",
        requiredEvents = Set.apply(
          PaymentReceived.event.name,
          FraudCheckCompleted.event.name
        )
      )
    )
}

File: examples/docs-code-snippets/src/main/java/examples/java/recipes/RecipeWithCheckpointEvent.java
package examples.java.recipes;

import com.ing.baker.recipe.javadsl.CheckPointEvent;
import com.ing.baker.recipe.javadsl.Recipe;
import examples.java.events.FraudCheckCompleted;
import examples.java.events.PaymentReceived;

public class RecipeWithCheckpointEvent {

    public final static Recipe recipe = new Recipe("example")
        .withCheckpointEvent(
            new CheckPointEvent("CheckpointReached")
                .withRequiredEvents(
                        PaymentReceived.class,
                        FraudCheckCompleted.class
                )
        );
}
